src/System.IO.FileSystem.DriveInfo/src/System.IO.FileSystem.DriveInfo.csproj

```diff
@@ -19,6 +19,12 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Linux_Release|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Debug|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU' " />
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU' "/>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU' "/>
+  <PropertyGroup>
+    <TargetsBSD Condition="'$(TargetsOSX)' == 'true' or '$(TargetsFreeBSD)' == 'true'">true</TargetsBSD>
+  </PropertyGroup>
```

Comment:
I think we have a naming problem here. BSD has a well defined meaning, referring to the open source OSes that derive from the original BSD release. That means FreeBSD, NetBSD, OpenBSD, DargonFly BSD etc. OS X is not a BSD, as we seem to agree.

Still that already defined term is used here in a different way, to refer to an API originating from the BSD family, not the OSes themselves. If at all, at least it should never be "TargetsBSD" but rather "TargetsBSDAPI". The latter makes clear we are speaking about API classification, not platform classification. Still better would be to just remove the "TargetsBSD" all together and below just include the individual files in the "TargetsFreeBSD" and "TargestOSX" sections.

Still I would strongly suggest to not add another folder in the Interop folder. Rather put those files in the Unix folder and name them as *.BSDAPI.cs.

We really must be clear and precise what exactly we are talking about. Using a term like BSD in this way is not really helping.

With the above points taken into account, I am fine with having implementations named by API families. But we really have to be clear and do not mix platform names and API family names in a confusing manner like it happened here.